src/components/inputs/localized-money-input/localized-money-input.js

```diff
@@ -0,0 +1,319 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import without from 'lodash.without';
+import oneLine from 'common-tags/lib/oneLine';
+import { injectIntl } from 'react-intl';
+import Spacings from '../../spacings';
+import Constraints from '../../constraints';
+import {
+  createLocalizedDataAttributes,
+  getHasErrorOnRemainingLanguages,
+  getHasWarningOnRemainingLanguages,
+  getId,
+  getName,
+} from '../../../utils/localized';
+import createSequentialId from '../../../utils/create-sequential-id';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+import MoneyInput from '../money-input';
+import CurrencyControl from './currency-control';
+import styles from './localized-money-input.mod.css';
+
+const sequentialId = createSequentialId('localized-money-input-');
+// sorts the currencies with the following priority:
+// - The selected currency is placed first (e.g EUR)
+// - All other currencies follow, sorted alphabetically as well
+export const sortCurrencies = (selectedCurrency, allCurrencies) => {
+  const remainingCurrencies = without(allCurrencies, selectedCurrency);
+  return [selectedCurrency, ...remainingCurrencies.sort()];
+};
+
+class LocalizedInput extends React.Component {
```

Comment:
Should this not be called `LocalizedMoneyInput`?

Reply:
This is the single input which then gets reused to build the `LocalizedMoneyInput`. We do the same in `LocalizedTextInput`. It's fine this way (although in hindsight `MoneyInput` would have been a more appropriate name - but then it gets confusing with the `MoneyInput` we already have; so this might actually be the least confusing name).

Reply:
Thanks for clarifying just wasn't sure and wanted to leave a note. Resolving.